src/matlab2tikz.m

```diff
@@ -3569,8 +3561,20 @@ function warnFacetedInterp(m2t)
                        env, drawOpts, tabOpts, table);
 end
 % ==============================================================================
+function marker = getMarkerInfo(m2t, h)
+    % gets marker-related options as a struct
+    marker                        = struct();
+    marker.style                  = get(h, 'Marker');
+    marker.FaceColor              = get(h, 'MarkerFaceColor');
+    marker.EdgeColor              = get(h, 'MarkerEdgeColor');
+    marker.hasFaceColor           = ~isNone(marker.FaceColor);
+    marker.hasEdgeColor           = ~isNone(marker.EdgeColor);
+    [marker.tikz, marker.options] = translateMarker(m2t, marker.style, ...
+        opts_new(), marker.hasFaceColor);
```

Comment:
Again the tiniest of nitpicks, can you put marker.options = opts_new();  on a single line and then pass marker.options into translateMarker?

I am just against passing functions as arguments